File: 00. Estructura/01. Lenguajes de proposito general/04. Paradigmas de programación/02. Orientada a objetos/10. Clases internas/Java/Clases internas.java
package Prueba;

// Crear clase con clases internas
class Clase {
    // Contenido
    class SubClase {
        // Contenido
        class SubSubClase {
            // Contenido
        }
        // Contenido
        SubSubClase subSubClase = new SubSubClase();     // Intanciación interior de subsubclase
        // Contenido
    }
    // Contenido
    SubClase subClase = new SubClase();                             // Intanciación interior de subclase
    SubClase.SubSubClase subSubClase = subClase.new SubSubClase();  // Intanciación exterior de subSubClase
    // Contenido
}

public class Prueba {

    public static void main(String[] args) {

        // Intanciación exterior de clase
        Clase clase = new Clase();
        
        // Intanciación exterior de subclase
        Clase.SubClase subClase = clase.new SubClase();
        
        // Intanciación exterior de subsubclase
        Clase.SubClase.SubSubClase subSubClase = subClase.new SubSubClase();

    }

}

File: 00. Estructura/01. Lenguajes de proposito general/04. Paradigmas de programación/02. Orientada a objetos/10. Clases internas/JavaScript/Clases internas.js

// Crear clase con clases internas
class Clase {
    // Contenido
    SubClase = class {
        // Contenido
        SubSubClase = class {
            // Contenido
        }
        // Contenido
        metodo() {
            // Contenido
            this.subsubClase = new this.SubSubClase();    // Intanciación interior de subsubclase
            // Contenido
        }
        // Contenido
    }
    // Contenido
    metodo() {
        // Contenido
        this.subClase = new this.SubClase();                  // Intanciación interior de subclase
        this.subsubClase = new this.subClase.SubSubClase();   // Intanciación interior de subsubclase
        // Contenido
    }
    // Contenido
}

// Intanciación exterior de clase
clase = new Clase();

// Intanciación exterior de subclase
subClase = new clase.SubClase();

// Intanciación exterior de subsubclase
subSubClase = new subClase.SubSubClase();
